h2o-py/h2o/frame.py

```diff
@@ -1977,7 +1977,8 @@ def as_data_frame(self, use_pandas=True, header=True):
                     elif can_use_polars() and can_use_pyarrow():  # polar/pyarrow if datatable is not available
                         return self.convert_with_polars(exportFile.name)
             warnings.warn("converting H2O frame to pandas dataframe using single-thread.  For faster conversion using"
-                          " multi-thread, install datatable, or polars and pyarrow.")
+                          " multi-thread, install datatable (for Python 3.9 or lower), or polars and pyarrow "
+                          "(for Python 3.10 or above).")
```

Comment:
Are we almost always printing this warning when converting to `pandas`.
This is probably all over the place now for users not having any of those libraries, especially with `as_data_frame()` also being used internally, this can become quickly annoying for them.

To avoid this, it is recommended to use a custom warning class, see `H2ODeprecationWarning` for an example.
You can add the following to `exceptions.py`:
```python
class H2ODependencyWarning(Warning):
    pass
```
and then on top of `h2o.py`:
```python
warnings.filterwarnings("once", category=H2ODependencyWarning)
```
then if you print your warning this way, it will be printed only on the first execution:
```python
 warnings.warn("converting H2O frame to pandas dataframe using single-thread.  For faster conversion using"
               " multi-thread, install datatable (for Python 3.9 or lower), or polars and pyarrow "
               "(for Python 3.10 or above).", H2ODependencyWarning)
```